include/mbedtls/check_config.h

```diff
@@ -720,6 +720,10 @@
 #error "MBEDTLS_SSL_DTLS_HELLO_VERIFY  defined, but not all prerequisites"
 #endif
 
+#if defined(MBEDTLS_SSL_DTLS_HELLO_VERIFY) && !defined(MBEDTLS_SSL_SRV_C)
```

Comment:
It looks to me like this should be in upstream Mbed TLS, too, but perhaps I'm missing something. I will open a PR and try to get feedback from the team.